### PR TITLE
Puts Index-CI tests in openshift branch

### PR DESCRIPTION
**Pull request by navidshaikh**
This changeset brings the index-CI tests framework and tests
 in OpenShift branch.

### CODE REVIEW

File: ci/cccp_ci.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
+# This script uses the Duffy node management api to get fresh machines to run
```
Comment (bamachrn) [inline ci/cccp_ci.py]: This file is not part of index-ci

File: ci/cccp_ci_cleanup.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import os
```
Comment (bamachrn) [inline ci/cccp_ci_cleanup.py]: File not part of index CI

File: ci/constants.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import os
```
Comment (bamachrn) [inline ci/constants.py]: not part of index-ci

File: ci/devcloud_ci.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python
```
Comment (bamachrn) [inline ci/devcloud_ci.py]: not part of index-ci

File: ci/lib.py [hunk 5]
```diff
@@ -0,0 +1,594 @@
+import json
```
Comment (bamachrn) [inline ci/lib.py]: not part of index-ci

File: ci/setup_env.sh [hunk 6]
```diff
@@ -0,0 +1,8 @@
+if [ -f env.properties ]; then
```
Comment (bamachrn) [inline ci/setup_env.sh]: not part of index-ci

File: ci/tests/README.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
+## Tests
```
Comment (bamachrn) [inline ci/tests/README.md]: not part of index-ci

File: ci/tests/base.py [hunk 8]
```diff
@@ -0,0 +1,134 @@
+import copy
```
Comment (bamachrn) [inline ci/tests/base.py]: not part of index-ci

File: ci/tests/test_01_functional/test_00_openshift/__init__.py [hunk 9]
```diff
@@ -0,0 +1,152 @@
+import time
```
Comment (bamachrn) [inline ci/tests/test_01_functional/test_00_openshift/__init__.py]: not part of index-ci

File: ci/tests/test_01_functional/test_02_linter/__init__.py [hunk 10]
```diff
@@ -0,0 +1,142 @@
+# module to test linter functionality
```
Comment (bamachrn) [inline ci/tests/test_01_functional/test_02_linter/__init__.py]: not part of index-ci

File: ci/tests/test_01_functional/test_03_scanner/__init__.py [hunk 11]
```diff
@@ -0,0 +1,79 @@
+# module to test scanner functionality
```
Comment (bamachrn) [inline ci/tests/test_01_functional/test_03_scanner/__init__.py]: not part of index-ci

File: ci/tests/test_01_functional/test_04_index_ci/__init__.py [hunk 12]
```diff
@@ -0,0 +1,253 @@
+from os import path, mkdir, utime
```
Comment (bamachrn) [inline ci/tests/test_01_functional/test_04_index_ci/__init__.py]: mostly not part of index part

File: ci/vendors/beanstalkc.py [hunk 13]
```diff
@@ -0,0 +1,315 @@
+#!/usr/bin/env python
```
Comment (bamachrn) [inline ci/vendors/beanstalkc.py]: not part of index-ci

**Comment by mohammedzee1000** [comment]
@navidshaikh maybe we can move the tests in separately?

**Comment by navidshaikh** [comment]
> @navidshaikh maybe we can move the tests in separately?

How does this help ?

**Comment by mohammedzee1000** [comment]
> > @navidshaikh maybe we can move the tests in separately?
> 
> How does this help ?

We might not need to move everything in here, and only pick what is relevant for index ci? So we should either make that now, or get index ci and and then move the tests portion in

**Pull request closed without merging**